lib/datasource/docker.js

```diff
@@ -24,12 +25,18 @@ function getRepository(pkgName) {
 
 async function getAuthHeaders(registry, repository) {
   // istanbul ignore if
-  if (registry !== 'https://index.docker.io') {
-    return {};
-  }
   try {
-    const authUrl = `https://auth.docker.io/token?service=registry.docker.io&scope=repository:${repository}:pull`;
-    logger.debug(`Obtaining docker registry token for ${repository}`);
+    const apiCheckUrl = `${registry}/v2/`;
+    const apiCheckResponse = await got(apiCheckUrl, { throwHttpErrors: false });
+    const authenticateHeader = new wwwAuthenticate.parsers.WWW_Authenticate(
+      apiCheckResponse.headers['www-authenticate']
+    );
+
+    // prettier-ignore
+    const authUrl = `${authenticateHeader.parms.realm}?service=${authenticateHeader.parms.service}&scope=repository:${repository}:pull`;
```

Comment:
Is it really parms and not params?

Reply:
Yes, this is the interface of the library.